From 477b38716186f5e1fd9149a4863b9e374f0a8091 Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Thu, 7 Jul 2022 15:44:20 +0200
Subject: [PATCH] arm64: dts: st: add scmi performance support on stm32mp25

Update the device-tree to add support for scmi performance
on cpu0 and cpu1. This leads to enable cpu-freq on the CPUs.

Signed-off-by: Pascal Paillet <p.paillet@st.com>
Change-Id: I853c03c86900c5959e93f42534a5b8bd72cf6ea0
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 7 +++++++
 arch/arm64/boot/dts/st/stm32mp253.dtsi | 2 ++
 2 files changed, 9 insertions(+)

--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -21,6 +21,8 @@
 			device_type = "cpu";
 			reg = <0>;
 			enable-method = "psci";
+			clocks = <&scmi_perf 0>;
+			clock-names = "cpu";
 		};
 	};
 
@@ -63,6 +65,11 @@
 			#size-cells = <0>;
 			linaro,optee-channel-id = <0>;
 
+			scmi_perf: protocol@13 {
+				reg = <0x13>;
+				#clock-cells = <1>;
+			};
+
 			scmi_clk: protocol@14 {
 				reg = <0x14>;
 				#clock-cells = <1>;
--- a/arch/arm64/boot/dts/st/stm32mp253.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp253.dtsi
@@ -12,6 +12,8 @@
 			device_type = "cpu";
 			reg = <1>;
 			enable-method = "psci";
+			clocks = <&scmi_perf 0>;
+			clock-names = "cpu";
 		};
 	};
 
